amp/WEB-INF/src/org/digijava/module/aim/util/LocationSkeleton.java

```diff
@@ -6,38 +6,66 @@
 
 /*
  * skeleton class for AmpCategoryValueLocation
- * 
+ *
  * */
 public class LocationSkeleton extends HierEntitySkeleton<LocationSkeleton> {
     protected Long cvId; // category value id
+    protected Long templateId;
+    protected Double lat;
+    protected Double lon;
 
-    public LocationSkeleton(Long id, String locName, String code, Long parentId, Long cvId) {
-        super(id, locName, code, parentId);
-        this.cvId = cvId;
+    public LocationSkeleton(LocationSkeletonBuilder locationSkeletonBuilder) {
```

Comment:
It is avoidable to fetch the build in the instance constructor.
The pattern is that the builder creates the instance.